ocaml/typing/persistent_env.ml

```diff
@@ -58,6 +58,7 @@ type can_load_cmis =
 
 type pers_struct = {
   ps_name: CU.t;
+  ps_is_param: bool;
```

Comment:
Why is this a bool rather than a `cmi_kind`?

Reply:
In later patches, `Cmi_format` goes into a correct-by-construction style: the `cmi_arg_for` field only makes sense for `Normal` modules, so for instance as of #1841, `cmi_arg_for` is a field of the `Normal` constructor of `cmi_kind`. For `Persistent_env`'s internal data structures, this seemed much less valuable, so `is_param` stays as a boolean.

(If we change #1841 and #1846 to keep `cmi_kind` a pure two-value enum, I'm happy to change `is_param`'s type to `cmi_kind` and rename it, probably to `cmi_kind` since the word "kind" is so overloaded (maybe I should change `cmi_kind` as well?).)